From 0028555d156ccb85e05e2711d119a3b4f924b779 Mon Sep 17 00:00:00 2001
From: Frank Wunderlich <frank-w@public-files.de>
Date: Sun, 3 Nov 2024 11:18:31 +0100
Subject: [PATCH 064/116] arm64: dts: update bpi-r4.dtsi to actual state

squashed:
arm64: dts: mediatek: mt7988a-bpi-r4: fix some binding errors
---
 .../dts/mediatek/mt7988a-bananapi-bpi-r4.dts  |  1 +
 .../dts/mediatek/mt7988a-bananapi-bpi-r4.dtsi | 31 +++++++++++++++++--
 2 files changed, 30 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4.dts b/arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4.dts
index 4b3796ba82e3..499f0c91c213 100644
--- a/arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4.dts
+++ b/arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4.dts
@@ -27,6 +27,7 @@ &gmac1 {
 	managed = "in-band-status";
 	phy-mode = "usxgmii";
 	sfp = <&sfp2>;
+	status = "okay";
 };
 
 &pca9545 {
diff --git a/arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4.dtsi b/arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4.dtsi
index 0ff69dae45d3..ddd8abee413d 100644
--- a/arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4.dtsi
+++ b/arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4.dtsi
@@ -4,8 +4,8 @@
 
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/leds/common.h>
+#include <dt-bindings/input/input.h>
 #include <dt-bindings/regulator/richtek,rt5190a-regulator.h>
-#include <dt-bindings/leds/common.h>
 
 #include "mt7988a.dtsi"
 
@@ -14,6 +14,10 @@ aliases {
 		ethernet0 = &gmac0;
 		ethernet1 = &gmac1;
 		ethernet2 = &gmac2;
+		led-boot = &led_green;
+		led-failsafe = &led_green;
+		led-running = &led_green;
+		led-upgrade = &led_green;
 	};
 
 	chosen {
@@ -29,6 +33,16 @@ fan: pwm-fan {
 		status = "okay";
 	};
 
+	gpio-keys {
+		compatible = "gpio-keys";
+
+		button-wps {
+			label = "WPS";
+			linux,code = <KEY_RESTART>;
+			gpios = <&pio 14 GPIO_ACTIVE_LOW>;
+		};
+	};
+
 	gpio-leds {
 		compatible = "gpio-leds";
 
@@ -152,6 +166,7 @@ &gmac2 {
 	managed = "in-band-status";
 	phy-mode = "usxgmii";
 	sfp = <&sfp1>;
+	status = "okay";
 };
 
 &gsw_phy0 {
@@ -281,7 +296,7 @@ pca9545: i2c-mux@70 {
 		#address-cells = <1>;
 		#size-cells = <0>;
 
-		i2c@0 {
+		i2c@0 { //eeprom,rtc,ngff
 			#address-cells = <1>;
 			#size-cells = <0>;
 			reg = <0>;
@@ -289,12 +304,16 @@ i2c@0 {
 			pcf8563: rtc@51 {
 				compatible = "nxp,pcf8563";
 				reg = <0x51>;
+				//interrupts = <&pio 6 IRQ_TYPE_LEVEL_LOW>;
 				#clock-cells = <0>;
+				//status = "disabled";
 			};
 
 			eeprom@57 {
 				compatible = "atmel,24c02";
 				reg = <0x57>;
+				address-width = <8>;
+				pagesize = <8>;
 				size = <256>;
 			};
 
@@ -305,6 +324,14 @@ i2c_sfp1: i2c@1 {
 			#size-cells = <0>;
 			reg = <1>;
 		};
+
+		i2c_wifi: i2c@3 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <3>;
+
+			status = "disabled";
+		};
 	};
 };
 
-- 
2.39.5

